sys/drv/rtc.c

```diff
@@ -107,9 +107,9 @@ static int rtc_attach(device_t *dev) {
     dev, RT_IOPORTS, 0, IO_RTC, IO_RTC + IO_RTCSIZE - 1, IO_RTCSIZE, RF_ACTIVE);
   assert(rtc->regs != NULL);
 
-  rtc->intr_handler =
-    INTR_HANDLER_INIT(rtc_intr, NULL, rtc, "RTC periodic timer", 0);
-  bus_intr_setup(dev, 8, &rtc->intr_handler);
+  /* intr_handler name: "RTC periodic timer" */
+  rtc->irq_res = bus_alloc_irq(dev, 0, 8 /* magic */, RF_ACTIVE);
```

Comment:
Maybe putting these IRQs into a header file shared between all drivers which do _this_ would be a good idea?

Reply:
I don't think that has much sense. When properly abstracted those device drivers will be able to work when attached to any bus (not necessarily ISA, refer to `ns16550` attached to `cbus`). Number 8 strictly belongs to ISA bus provided by GT64120. AFAIK interrupts number on x86 may be provided by ISA PNP or ACPI. In our case we want FDT to take this reponsibility.

Reply:
I'd still prefer a separate file that would just have a bunch of defines for these as a **temporary** solution. It would be easier to tell if there are any unused IRQs or if there's a conflict. But I guess this is good enough for now as we are unlikely to mess with new interrupts before introducing _resource lists_ or some other mechanism that will allow a bus to pass IRQs to drivers. Also, the `gt_pci_intr_name` that's now defined in _gt64120.c_ sort of already lists the interrupts and their usage.